src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy

```diff
@@ -0,0 +1,1189 @@
+            config.auction.priceFloors.fetch.url = null
+        }
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS should log error"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, bidRequest.site.publisher.id)
+        assert floorsLogs.size() == 1
+        assert floorsLogs == ["placeholder"]
+
+        and: "#metric should be update"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid.isEmpty()
+
+        and: "PBS should fall back to the startup configuration"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.imp[0]?.bidFloor == bidRequest.imp[0].bidFloor
+    }
+
+    def "PBS should validate fetch.max-age-sec from account config"() {
+        given: "Pbs with PF configuration with minMaxAgeSec"
+        def minMaxAgeSec = PBSUtils.randomNumber
+        def defaultAccountConfigSettings = defaultAccountConfigSettings.tap {
+            auction.priceFloors.fetch.minMaxAgeSec = minMaxAgeSec
+        }
+        def pbsService = pbsServiceFactory.getService(floorsConfig +
+                ["settings.default-account-config": mapper.encode(defaultAccountConfigSettings)])
+
+        and: "Test start time"
+        def startTime = Instant.now()
+
+        and: "Flush metric #metric"
+        def metric = "placeholder"
+        flushMetrics()
+
+        and: "Default BidRequest"
+        def bidRequest = BidRequest.getDefaultBidRequest(APP)
+
+        and: "Account with enabled fetch, fetch.url, maxAgeSec in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id).tap {
+            config.auction.priceFloors.fetch.maxAgeSec = minMaxAgeSec - 1
+        }
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS log should error"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, bidRequest.site.publisher.id)
+        assert floorsLogs.size() == 1
+        assert floorsLogs == ["placeholder"]
+
+        and: "#metric should be update"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid.isEmpty()
+    }
+
+    def "PBS should validate fetch.period-sec from account config"() {
+        given: "Pbs with PF configuration with minMaxAgeSec, maxAgeSec"
+        def minMaxAgeSec = PBSUtils.randomNumber
+        def maxAgeSec = PBSUtils.randomNumber + 10
```

Comment:
This doesn't seem right? If the random number is different on each line, then minMaxAgeSec might be larger or smaller than the maxAge. Seems like we want it consistent within the test. Like "def maxAgeSec=minMaxAgeSec+10"